src/ServiceControlInstaller.Engine/FileSystem/FileUtils.cs

```diff
@@ -16,26 +17,60 @@ public static string SanitizeFolderName(string folderName)
 
         public static void DeleteDirectory(string path, bool recursive, bool contentsOnly, params string[] excludes)
         {
-            if (!Directory.Exists(path))
+            var di = new DirectoryInfo(path);
+
+            if (!di.Exists)
+            {
+                return;
+            }
+
+            var originalPath = path;
+            var pathToDelete = path + "$";
+
+            try
+            {
+                // Move folder to ensure no files are in use so that it is less likely that we corrupt the folder
+                // when still in use
+                RunWithRetries(() => di.MoveTo(pathToDelete));
+                path = pathToDelete;
+                DeleteDirectoryInternal(path, recursive, contentsOnly, excludes);
```

Comment:
Added linux timestamp in seconds instead of uuid to not make the file much longer.

Could alternatively just rename the whole name. That likely could be a uuid as it would be 36 characters where very often the name would be something like `Particular.ServiceControl-DEV` which is a little less.

What would you prefer @johnsimons ?

Reply:
`Path.GetRandomName()` - https://learn.microsoft.com/en-us/dotnet/api/system.io.path.getrandomfilename?view=net-8.0

Reply:
nice api! Didn't know that one. Added a clarifying comment that adding to the folder is intended